smac_planner/test/test_costmap_downsampler.cpp

```diff
@@ -31,40 +30,115 @@ class RclCppFixture
 };
 RclCppFixture g_rclcppfixture;
 
+class TestNode : public nav2_util::LifecycleNode
```

Comment:
Can you explain your thinking a little on this test node? I'm not sure I see the benefit of these changes

Reply:
This is the last point of just trying to understand the need behind the changes :smile:

Reply:
Ah yes on second look seems like those changes were redundant. I was under the impression that the test was failing on cyclonedds because the parent `nav2_util::LifecycleNode` needed to be in the configuring lifecycle transition state when the downsampler is configured. I've reverted the changes in the latest commit and everything seems to work now.